### PR TITLE
Remove redundant cc score code

### PR DESCRIPTION
**Pull request by mumichae**
A fresh version of PR #219

### CODE REVIEW

File: scIB/metrics.py [hunk 1]
```diff
@@ -558,97 +561,61 @@ def cell_cycle(adata_pre, adata_post, batch_key, embed=None, agg_func=np.mean,
...
-            int_sub = int_sub.obsm[embed] if embed is not None else int_sub.X
-        
-            if raw_sub.shape[0] != int_sub.shape[0]:
-                message = f'batch "{batch}" of batch_key "{batch_key}" '
-                message += 'has unequal number of observations before and after integration.'
-                message += f'before: {raw_sub.shape[0]} after: {int_sub.shape[0]}'
-                raise ValueError(message)
-        
+
+    recompute_cc = recompute_cc or ('S_score' not in adata_pre.obs_keys() and 'G2M_score' not in adata_pre.obs_keys())
```
Comment (LuckyMD) [inline scIB/metrics.py:565]: should this not be an `or` in there? We still can't compute scores if only one of `'S_score'` or `'G2M_score'` are there.

**Review by LuckyMD — CHANGES_REQUESTED** [review]
one minor change. And is this tested for consistent results? Should be all the same code by the looks of things, but would be good to confirm once with precomputed stuff and once without.

File: scIB/metrics.py (hunk 1 above)
Comment (mumichae) [reply scIB/metrics.py:565]: True, then it'd better be
```py
recompute_cc = recompute_cc or 'S_score' not in adata_pre.obs_keys() or 'G2M_score' not in adata_pre.obs_keys()
```

**Review by LuckyMD — APPROVED** [review]
Could you confirm that this is tested for consistency on test data before merging though?

**Comment by mumichae** [comment]
I've tested it for a couple of methods and the results are identical for this branch and the master branch

**Pull request merged**